app/client/jsx/Welcome.jsx

```diff
@@ -34,23 +46,28 @@ class Welcome extends Component {
     }
 
     async handleReady() {
-        console.log(this.state)
         const response = await this.httpApi.join(this.state.displayName, this.state.avatar)
         if (response.success) {
             const { displayName, avatar } = this.state
-            this.props.updateUser({
+            const user = {
                 displayName,
                 avatar,
                 userId: response.userId
-            })
-            this.props.updateCurrentRoom({
-                room: 'vestibule',
-                entered: false
-            })
-            this.setState({ redirect: '/party' })
+            }
+            this.handleUserSelected(user)
+            this.setState({redirect: '/party'})
         }
     }
 
+    handleUserSelected(user) {
+        LocalStorage.set("USER", user)
```

Comment:
So this only sets user in local storage once "party" is clicked, right?

Reply:
yes, this is called either when "party" is clicked or when someone with a "session" hits the welcome page